monkey/infection_monkey/island_api_client/configuration_validator_decorator.py

```diff
@@ -34,6 +34,13 @@ def get_agent_binary(self, operating_system: OperatingSystem) -> bytes:
     def get_agent_plugin(self, plugin_type: AgentPluginType, plugin_name: str) -> AgentPlugin:
         return self._island_api_client.get_agent_plugin(plugin_type, plugin_name)
 
+    def get_agent_plugin_manifest(
+        self,
+        plugin_type: AgentPluginType,
+        plugin_name: str,
+    ) -> AgentPluginManifest:
+        return self._island_api_client.get_agent_plugin_manifest(plugin_type, plugin_name)
```

Comment:
Every time `IIslandAPIClient` changes (which should be quite often) we'll have to add boilerplate code like this.
I've played around with the code for a bit and found a way to avoid the boilerplate code. What do you think? @mssalvatore  
```
def decorator(cls):
    class Wrapper(cls):
        def get_name(self):
            # fetches the name attribute
            return self.name
    return Wrapper
    
@decorator
class C:
    def __init__(self, y, x):
        self.name = y
        self.surname = x
    def get_surname(self):
        return self.surname
    def get_name(self):
        return "this should be overridden"
  
```

Then you can use it:
```
a = C("vakari", "is") -> 
a.get_surname() -> 'is'
a.get_name() -> 'vakari'
```

Reply:
There are a few ways to do it, each one has some issues or limitations imposed by the language.

We can revisit this next release. For now it's not a priority.